### DISCUSSION TITLE
Hover effect implementation

### DISCUSSION THREAD

**Discussion by SolarGuy8** [katspaugh/wavesurfer.js#3478] [Q&A] [answered]
Hi! 

I want to implement hover effect (same as in SoundCloud). I found Soundcloud-like player in Doc, but hover effect reached using the div and if you will open Browser console, you will see, that it's just rectrange div. I want to implement a hover div with the same form as waveform.

Maybe someone already implement this? If not, then could you help me, how can I change the wave color from point A to point B only. Or how can I get the reference to canvas element (using React.js). Thank you in Advance.

![ScreenRecording2024-01-09at21 47 36-ezgif com-video-to-gif-converter (1)](https://github.com/katspaugh/wavesurfer.js/assets/36670965/6e1c2930-4b74-4795-a01f-b379c425921a)

**Comment by katspaugh** [comment, marked as answer]
This can be done with a div overlay with a filter:

```css
.overlay {
  background: #756752;
  mix-blend-mode: 'plus-lighter';
}
```

You could extend the Hover plugin with this, it would be a cool addition.

**Reply by SolarGuy8** [reply]
<img width="754" alt="Screenshot 2024-01-10 at 17 05 45" src="https://github.com/katspaugh/wavesurfer.js/assets/36670965/667c8652-a186-4149-8f63-cd4a6e79ade6">

Seems I still the the border of div 🤔
Maybe, you can help me?

**Reply by SolarGuy8** [reply]
I don't understand, why my hover color transform into `rgb(56, 56, 56)` instead of `rgb(28, 28, 28)`, while I set, that background color should be `rgb(28, 28, 28)` even with `!important` option

**Reply by katspaugh** [reply]
Ah yes, sorry, the mix-blend-mode should be on another div that contains the actual overlay.
The overlay background should be `rgb(30, 30, 30)`, i.e. the same the background of the waveform.

<img width="972" alt="Screenshot 2024-01-10 at 15 18 13" src="https://github.com/katspaugh/wavesurfer.js/assets/381895/496f44a2-4eaf-499d-bf28-95dfe0182a48">

**Reply by SolarGuy8** [reply]
You mean create a parent div for hover? Or div inside the hover?

```
<div className={classes.root}>
      <div id="waveform">
        <div>
          <div id="hover"></div>
        </div>
      </div>
</div>
```

and my css
```
      '& #hover': {
        position: 'absolute',
        left: 0,
        top: 0,
        zIndex: 10,
        pointerEvents: 'none',
        height: '55%',
        width: 0,
        opacity: 0,
        transition: 'opacity 0.2s ease',
        background: 'rgba(28, 28, 28)',
        mixBlendMode: 'plus-lighter',
      },

      '& #waveform': {
        cursor: 'pointer',
        position: 'relative',
      },

      '& #waveform:hover #hover': {
        opacity: 1,
      },
```

**Reply by SolarGuy8** [reply]
Maybe I've done wrong, but I still have a troubles (@katspaugh)

```
    <div className={classes.root}>
      <div id="waveform">
        <div id="hoverWrapper">
          <div id="hover"></div>
        </div>
      </div>
    </div>
```

and styles:

```
    root: {
      position: 'relative',

      '& #hoverWrapper': {
        position: 'absolute',
        left: 0,
        top: 0,
        zIndex: 10,
        pointerEvents: 'none',
        height: '55%',
        width: '100%',
        transition: 'opacity 0.2s ease',
        mixBlendMode: 'plus-lighter',
      },

      '& #hover': {
        position: 'absolute',
        left: 0,
        top: 0,
        zIndex: 10,
        pointerEvents: 'all',
        height: '100%',
        width: 0,
        transition: 'opacity 0.2s ease',
        background: 'rgba(28, 28, 28)',
      },

      '& #waveform': {
        cursor: 'pointer',
        position: 'relative',
      },

      '& #waveform:hover #insideHover': {
        opacity: 1,
      },
    },
```

**Reply by SolarGuy8** [reply]
Just added z-index to waveform and everything start working :) Thank you!

**Reply by katspaugh** [reply]
Please feel free to add this functionality to the Hover plugin if you want. Cheers!

**Comment by baranowskyi** [comment]
I have a gradient background. And I couldn't make it like the example from soundcloud either. The suggested solutions don't work.
![hover](https://github.com/user-attachments/assets/1d7c60de-d0ff-4ef3-91e5-126488f05530)
